### DISCUSSION TITLE
How can I get point cloud data with LIDAR?

### DISCUSSION THREAD

**Discussion by psh9002** [Toni-SM/skrl#85] [Q&A]
Hi,

I try to use LIDAR in my environment but, the points are not created at the surface of the objects. Is there some considerations for using LIDAR?

![image](https://github.com/Toni-SM/skrl/assets/15624901/079b76b0-c4ad-4f7b-9076-e1e6d4152b83)
Above is what I want in RL environment. but....

![image](https://github.com/Toni-SM/skrl/assets/15624901/ea0c04cb-c7dd-4c34-a9ae-ef5c36f897ac)
There are no point clouds on the objects surface.

Here is my code.
[moving_target_point_cloud.zip](https://github.com/Toni-SM/skrl/files/11917238/moving_target_point_cloud.zip)

**Comment by Toni-SM** [comment]
Hi @psh9002 

I think the problem (visualization of lidar data) may come from the [fabric](https://docs.omniverse.nvidia.com/app_create/prod_extensions/ext_physics.html#physx-short-flatcache-also-known-as-fabric-rename-in-next-release) extension which is enabled as indicated in the top-left windows.

Fabric notes:

- Using Fabric means that the data are not available in USD. This will cause USD based functionality to not work. For example debug visualization of physics colliders wont work anymore, because the transformations are not synchronized.
- Using a python script to update and get USD state wont work because USD does not have current transformation information.

<hr>

Because it is not related to **skrl**, but to Isaac Sim, I will close this discussion. 
Please create a new topic in the [NVIDIA Isaac Sim forum](https://forums.developer.nvidia.com/c/omniverse/simulation/69) to get better support.
